### PR TITLE
add inbound fees fields to channel event

### PR DESCRIPTION
**Pull request by bufo24**
Adds needed fields to be compatible with the new fields added in this PR:

https://github.com/lightningnetwork/lnd/pull/8723

Will change from draft to ready for review whenever the LND PR gets merged

### CODE REVIEW

File: lnd_responses/rpc_channel_update_as_update.js [hunk 1]
```diff
@@ -20,6 +20,8 @@ const now = () => new Date().toISOString();
       disabled: <Forwarding is Disabled Bool>
       fee_base_msat: <Forwarding Base Fee Millitokens String>
       fee_rate_milli_msat: <Forwarding Fee Rate Per Million String>
+      inbound_fee_base_msat: <Inbound Forwarding Base Fee Millitokens Number>
+      inbound_fee_rate_milli_msat: <Inbound Forwarding Fee Rate Per Million Number>
```
Comment (alexbosworth) [inline lnd_responses/rpc_channel_update_as_update.js:24]: This will be a string

**Comment by alexbosworth** [comment]
I was thinking about calling this source_discount rather than inbound_fee to give more clarity towards what it is typically doing, reducing the outbound fee rate based on the source

**Comment by bufo24** [comment]
I know LND also calls it inbound fees, so it's nice to be consistent with the naming, so other devs directly understand that we're talking about the same thing, instead of having to figure out it source_discount and inbound fees are the same.

**Comment by alexbosworth** [comment]
I don't think it would be the same because I'd flip the sign on it, although I have to play with it to know

Consistency with naming isn't a goal of the arguments generally

File: lnd_responses/rpc_channel_update_as_update.js [hunk 2]
```diff
@@ -125,12 +125,12 @@ module.exports = update => {
     cltv_delta: update.routing_policy.time_lock_delta,
     fee_rate: Number(update.routing_policy.fee_rate_milli_msat),
     id: chanFormat({number: update.chan_id}).channel,
-    inbound_base_fee_mtokens: update.inbound_fee_base_msat,
-    inbound_fee_rate: update.routing_policy.inbound_fee_rate_milli_msat,
     is_disabled: update.routing_policy.disabled,
     max_htlc_mtokens: maxHtlc !== Number().toString() ? maxHtlc : undefined,
     min_htlc_mtokens: update.routing_policy.min_htlc,
     public_keys: [update.advertising_node, update.connecting_node],
+    source_discount_base_mtokens: Number(update.inbound_fee_base_msat),
```
Comment (alexbosworth) [inline lnd_responses/rpc_channel_update_as_update.js:132]: this can still be string

**Comment by alexbosworth** [comment]
I'm playing around with this on the other APIs, just a heads up not 100% settled on naming but I can adjust it

**Comment by alexbosworth** [comment]
OK what I settled on is:

`inbound_fee_base_msat` should map to `inbound_base_discount_mtokens` but negative sign (unless 0, then 0), and a string rather than a number

`inbound_fee_rate_milli_msat` should map to `inbound_rate_discount` but negative sign (unless 0, then just 0)

**Comment by bufo24** [comment]
And I tested this and I see that the inbound fees are coming in like numbers, not strings:
```
{
    custom_records: {
      '\x03Ù\x00\x00\x00\x00\x00\x00': <Buffer ff ff ff f6 ff ff ff ec>
    },
    time_lock_delta: 50,
    min_htlc: '1000',
    fee_base_msat: '10',
    fee_rate_milli_msat: '10000000',
    disabled: false,
    max_htlc_msat: '49500000',
    last_update: 0,
    inbound_fee_base_msat: -10,
    inbound_fee_rate_milli_msat: -20
  }
  ```

**Comment by alexbosworth** [comment]
Yeah I saw that too, I pushed some changes for the existing inbound fees that should show the transform

File: lnd_methods/info/subscribe_to_graph.d.ts [hunk 3]
```diff
@@ -19,6 +19,10 @@ export type SubscribeToGraphChannelUpdatedEvent = {
   min_htlc_mtokens: string;
   /** Announcing Public Key, Target Public Key */
   public_keys: [string, string];
+  /** Channel Source Discount Base Fee Millitokens */
+  source_discount_base_mtokens: string;
```
Comment (alexbosworth) [inline lnd_methods/info/subscribe_to_graph.d.ts:23]: need to be changed to new format?

File: lnd_methods/info/subscribe_to_graph.js [hunk 4]
```diff
@@ -49,6 +49,8 @@ const type = 'default';
     [max_htlc_mtokens]: <Channel Maximum HTLC Millitokens String>
     min_htlc_mtokens: <Channel Minimum HTLC Millitokens String>
     public_keys: [<Announcing Public Key>, <Target Public Key String>]
+    source_discount_base_mtokens: <Channel Base Fee Millitokens Number>
```
Comment (alexbosworth) [inline lnd_methods/info/subscribe_to_graph.js:52]: needs to use new format?

File: lnd_responses/rpc_channel_update_as_update.js [hunk 5]
```diff
@@ -20,6 +20,8 @@ const now = () => new Date().toISOString();
       disabled: <Forwarding is Disabled Bool>
       fee_base_msat: <Forwarding Base Fee Millitokens String>
       fee_rate_milli_msat: <Forwarding Fee Rate Per Million String>
+      inbound_fee_base_msat: <Inbound Forwarding Base Fee Millitokens Number>
```
Comment (alexbosworth) [inline lnd_responses/rpc_channel_update_as_update.js:23]: should be discount?

File: lnd_responses/rpc_channel_update_as_update.js [hunk 6]
```diff
@@ -117,16 +119,28 @@ module.exports = update => {
...
+    cltv_delta: time_lock_delta,
+    fee_rate: Number(fee_rate_milli_msat),
+    id: chanFormat({ number: update.chan_id }).channel,
+    is_disabled: disabled,
     max_htlc_mtokens: maxHtlc !== Number().toString() ? maxHtlc : undefined,
-    min_htlc_mtokens: update.routing_policy.min_htlc,
+    min_htlc_mtokens: min_htlc,
     public_keys: [update.advertising_node, update.connecting_node],
+    inbound_base_discount_mtokens: inbound_fee_base_msat.toString(),
+    inbound_rate_discount: inbound_fee_rate_milli_msat.toString(),
```
Comment (alexbosworth) [inline lnd_responses/rpc_channel_update_as_update.js:143]: the rate should be a number

File: lnd_responses/rpc_channel_update_as_update.js [hunk 7]
```diff
@@ -117,16 +119,28 @@ module.exports = update => {
 
   const txId = !!transactionId.equals(emptyTxId) ? null : transactionId;
 
+  const {
+    disabled,
```
Comment (alexbosworth) [inline lnd_responses/rpc_channel_update_as_update.js:123]: i don't think the other variables need to change

**Review by alexbosworth — CHANGES_REQUESTED** [review]
can consolidate around the new format like in getchannel

File: lnd_responses/rpc_channel_update_as_update.js (hunk 5 above)
Comment (bufo24) [reply lnd_responses/rpc_channel_update_as_update.js:23]: This is the incoming object from lnd, don't think they mention `discount`

**Pull request merged**